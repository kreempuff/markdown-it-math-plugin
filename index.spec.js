let MarkdownIt = require("markdown-it");
let mylib = require("./index");

let md;
describe('main', function () {
  beforeEach(() => {
    md = new MarkdownIt();
    md.use(mylib)
  });
  test("test basic parsing", () => {
    let parsed = md.render("$hello$");
    expect(parsed).toMatchSnapshot()
  })

  test("test display math", () => {
    let parsed = md.render("$$hello$$");
    expect(parsed).toMatchSnapshot()
  })
});
