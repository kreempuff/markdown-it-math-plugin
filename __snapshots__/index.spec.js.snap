// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main test basic parsing 1`] = `
"<p><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mi>h</mi><mi>e</mi><mi>l</mi><mi>l</mi><mi>o</mi></mrow><annotation encoding=\\"application/x-tex\\">hello</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.69444em;vertical-align:0em;\\"></span><span class=\\"mord mathdefault\\">h</span><span class=\\"mord mathdefault\\">e</span><span class=\\"mord mathdefault\\" style=\\"margin-right:0.01968em;\\">l</span><span class=\\"mord mathdefault\\" style=\\"margin-right:0.01968em;\\">l</span><span class=\\"mord mathdefault\\">o</span></span></span></span></p>
"
`;

exports[`main test display math 1`] = `
"<p><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mi>h</mi><mi>e</mi><mi>l</mi><mi>l</mi><mi>o</mi></mrow><annotation encoding=\\"application/x-tex\\">hello</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.69444em;vertical-align:0em;\\"></span><span class=\\"mord mathdefault\\">h</span><span class=\\"mord mathdefault\\">e</span><span class=\\"mord mathdefault\\" style=\\"margin-right:0.01968em;\\">l</span><span class=\\"mord mathdefault\\" style=\\"margin-right:0.01968em;\\">l</span><span class=\\"mord mathdefault\\">o</span></span></span></span></span></p>
"
`;
